src/main/scala/com/evolution/scache/LoadingCache.scala

```diff
@@ -116,50 +118,92 @@ private[scache] object LoadingCache {
                             .attempt
                             .race1 { deferred.get }
                             .flatMap {
+                              // `value` got computed, and deferred was not (yet) completed by any other fiber in `put`
                               case Left(Right((a, entry))) =>
-                                0.tailRecM { counter =>
-                                  entryRef
-                                    .access
-                                    .flatMap {
-                                      case (Right(entry0), _) =>
-                                        entry
-                                          .release1
-                                          .as {
-                                            entry0
-                                              .value
-                                              .asRight[F[V]]
-                                              .asRight[A]
-                                              .asRight[Int]
-                                          }
+                                deferred
```

Comment:
You mean the order `entryRef set` and `deferred.complete`?
The reasoning here is to align the order with the one in `put`, although since we eliminated most of the retries, it doesn't really matter.
In `getOrUpdate1` we want to try to complete deferred in pretty much all the cases anyways, so might as well do it first.

Reply:
yes, please